packages/flutter/lib/src/rendering/object.dart

```diff
@@ -507,7 +510,10 @@ class PaintingContext extends ClipContext {
     assert(clipBehavior != null);
     final Rect offsetBounds = bounds.shift(offset);
     final Path offsetClipPath = clipPath.shift(offset);
-    if (needsCompositing) {
+    if (clipBehavior == Clip.none) {
```

Comment:
nit: I believe this will allow us to clean up some other call-sides that currently re-implement this branch and don't call pushClipPath if the behavior is none (e.g. RenderClipOval)

Reply:
Potentially the same for pushClipRRect.

Reply:
Done